package.json

```diff
@@ -0,0 +1,14 @@
+{
+  "name": "blockonomics-bitcoin-payments",
+  "version": "1.0.0",
+  "description": "**Contributors:** juhasiivikko, darrenwestwood, blockonomics **Tags:** bitcoin, accept bitcoin, bitcoin woocommerce, bitcoin wordpress plugin, bitcoin payments  **Requires at least:** 3.0.1  **Tested up to:** 6.3.1 **Stable tag:** 3.6.9 **License:** MIT  **License URI:** http://opensource.org/licenses/MIT",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "wp-scripts build js/block.js"
+  },
+  "devDependencies": {
+    "@woocommerce/settings": "^1.0.0",
```

Comment:
To improve dependency management, we can use `@woocommerce/blocks-registry` and `@woocommerce/settings` as dependencies. 
We can refer to the [woocommerce dummy gateway](https://github.com/woocommerce/woocommerce-gateway-dummy) which uses [@woocommerce/dependency-extraction-webpack-plugin](https://github.com/woocommerce/woocommerce-gateway-dummy/blob/master/package.json#L13) and [Webpack](https://github.com/woocommerce/woocommerce-gateway-dummy/blob/master/webpack.config.js) to allow `@woocommerce/blocks-registry` and `@woocommerce/settings` to be [included as dependencies](https://github.com/woocommerce/woocommerce-gateway-dummy/blob/master/resources/js/frontend/index.js#L2-L5).

This will also allow us to use wp-scripts with our checkout js to bundle our dependencies in the future.

Reply:
It is not necessary, but I think it will be a good idea to include this for uniformity. All dependencies will then be handled in the same manner.
wp-scripts uses @wordpress/dependency-extraction-webpack-plugin to manage Wordpress dependencies, so it makes to also use @woocommerce/dependency-extraction-webpack-plugin for Woocommerce dependencies.

**Wordpress**
```
@wordpress/react -> window.React
@wordpress/i18n -> window.wp.i18n
@wordpress/html-entities-> window.wp.htmlEntities
```

**Woocommerce**
```
@woocommerce/blocks-registry -> window.wc.wcBlocksRegistry
@woocommerce/settings -> window.wc.wcSettings
```

Reply:
Note we will require a webpack config file in order to enable the WooCommerceDependencyExtractionWebpackPlugin to run during `wp-scripts build`. eg.
```
const defaultConfig = require('@wordpress/scripts/config/webpack.config');
const WooCommerceDependencyExtractionWebpackPlugin = require('@woocommerce/dependency-extraction-webpack-plugin');
const path = require('path');

// Export configuration.
module.exports = {
	...defaultConfig,
	entry: './js/block.js',
	output: {
		path: path.resolve( __dirname, 'build' ),
		filename: 'block.js',
	},
	plugins: [
		...defaultConfig.plugins.filter(
			(plugin) =>
				plugin.constructor.name !== 'DependencyExtractionWebpackPlugin'
		),
		new WooCommerceDependencyExtractionWebpackPlugin()
	]
};
```